src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -103,18 +109,28 @@ await bindingContext.ValueProvider.ContainsPrefixAsync(bindingContext.ModelName)
 
         private async Task<bool> CanValueBindAnyModelProperties(MutableObjectBinderContext context)
         {
-            // We need to enumerate the non marked properties and properties marked with IValueProviderMetadata
-            // instead of checking bindingContext.ValueProvider.ContainsPrefixAsync(bindingContext.ModelName)
-            // because there can be a case where a value provider might be willing to provide a marked property,
-            // which might never be bound.
-            // For example if person.Name is marked with FromQuery, and FormValueProvider has a key person.Name,
-            // and the QueryValueProvider does not, we do not want to create Person.
+            // We want to check to see if any of the properties of the model can be bound using the value providers,
+            // because that's all that MutableObjectModelBinder can handle.
+            //
+            // To do this we need to enumerate the properties, and see which of them provide a binding source
+            // through metadata, then we decided what to do.
+            //
+            //      If a property has a binding source, and it's a greedy source, then it's not
```

Comment:
There is one more case, If all the properties have a greedy source, then it's container object is allowed to be created.